src/ports/postgres/modules/convex/mlp_igd.py_in

```diff
@@ -0,0 +1,754 @@
+# coding=utf-8
```

Comment:
Pylint rating (src/ports/postgres/modules/convex/mlp_igd.py_in): `Your code has been rated at 6.37/10`

```
PYTHONPATH=/usr/local/madlib/Versions/1.12-dev/ports/postgres/9.6/modules pylint ../src/ports/postgres/modules/convex/mlp_igd.py_in
No config file found, using default configuration
************* Module mlp_igd
W:152, 0: TODO validation for summary (fixme)
W:477, 0: TODO add join for validation (fixme)
W:488, 0: TODO (fixme)
W:669, 0: TODO (fixme)
C: 82, 0: Unnecessary parens after u'if' keyword (superfluous-parens)
C: 99, 0: Unnecessary parens after u'if' keyword (superfluous-parens)
C:149, 0: Exactly one space required around assignment
        layer_sizes_str=py_list_to_sql_string(layer_sizes, array_type="integer")
                       ^ (bad-whitespace)
C:150, 0: Exactly one space required around assignment
        classes_str=py_list_to_sql_string([strip_end_quotes(cl, "'") for cl in classes],
                   ^ (bad-whitespace)
C:251, 0: Unnecessary parens after u'if' keyword (superfluous-parens)
C:329, 0: Line too long (101/100) (line-too-long)
C:372, 0: Exactly one space required after comma
                       'tolerance', 'step_size','n_iterations',
                                               ^ (bad-whitespace)
C:385, 0: Exactly one space required around comparison
    is_response = int(pred_type=='response')
                               ^^ (bad-whitespace)
C:410, 0: Unnecessary parens after u'if' keyword (superfluous-parens)
C:430, 0: Line too long (116/100) (line-too-long)
C:458, 0: Wrong hanging indentation (remove 4 spaces).
                        for i, c in enumerate(classes)])
                    |   ^ (bad-continuation)
C:492, 0: Line too long (111/100) (line-too-long)
C:493, 0: Line too long (138/100) (line-too-long)
C:495, 0: Line too long (120/100) (line-too-long)
C:496, 0: Line too long (111/100) (line-too-long)
W:566, 0: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
E: 27, 0: Unable to import 'plpy' (import-error)
R: 46, 0: Too many arguments (9/5) (too-many-arguments)
R: 46, 0: Too many local variables (40/15) (too-many-locals)
C: 91,16: Invalid variable name "x" (invalid-name)
W:137,-1: Unused format argument 'tolerance' (unused-format-string-argument)
W: 46, 0: Unused argument 'kwargs' (unused-argument)
W:149, 8: Unused variable 'layer_sizes_str' (unused-variable)
W: 80, 8: Unused variable 'original_dependent_varname' (unused-variable)
W: 73, 8: Unused variable 'n_tries' (unused-variable)
W:150, 8: Unused variable 'classes_str' (unused-variable)
R: 46, 0: Too many statements (52/50) (too-many-statements)
W:189,65: Unused argument 'n_iterations' (unused-argument)
W:189,38: Unused argument 'output_table' (unused-argument)
W:190, 4: Unused variable 'final_state_str' (unused-variable)
W:189,23: Unused argument 'schema_madlib' (unused-argument)
R:230, 0: Too many arguments (6/5) (too-many-arguments)
W:230,48: Unused argument 'output_table' (unused-argument)
W:231,53: Unused argument 'hidden_layer_sizes' (unused-argument)
W:232,34: Unused argument 'optimizer_params' (unused-argument)
W:230,62: Unused argument 'independent_varname' (unused-argument)
R:243, 0: Too many arguments (6/5) (too-many-arguments)
W:243,44: Unused argument 'output_table' (unused-argument)
W:244,49: Unused argument 'hidden_layer_sizes' (unused-argument)
W:245,30: Unused argument 'optimizer_params' (unused-argument)
W:243,58: Unused argument 'independent_varname' (unused-argument)
R:257, 0: Too many arguments (7/5) (too-many-arguments)
C:283,12: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:285,16: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:340, 7: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
R:346, 0: Too many arguments (6/5) (too-many-arguments)
R:346, 0: Too many local variables (30/15) (too-many-locals)
W:483, 4: Unnecessary pass statement (unnecessary-pass)
W:382, 4: Unused variable 'activation_function' (unused-variable)
W:432,16: Unused variable 'classes_table' (unused-variable)
W:455,16: Unused variable 'score_format' (unused-variable)
W:387, 4: Unused variable 'pred_name' (unused-variable)
W:346,16: Unused argument 'schema_madlib' (unused-argument)
W:431,16: Unused variable 'classes_with_index_table' (unused-variable)
W:385, 4: Unused variable 'is_response' (unused-variable)
W:378, 4: Unused variable 'coeff' (unused-variable)
W:383, 4: Unused variable 'layer_sizes' (unused-variable)
W:407,12: Unused variable 'unnest_if_not_array' (unused-variable)
W:346, 0: Unused argument 'kwargs' (unused-argument)
C:486, 0: Missing function docstring (missing-docstring)
R:486, 0: Too many local variables (17/15) (too-many-locals)
C:493, 4: Invalid variable name "label_description_classification" (invalid-name)
R:657, 4: Unnecessary "else" after "return" (no-else-return)
C:668, 0: Missing function docstring (missing-docstring)
R:745, 4: Unnecessary "else" after "return" (no-else-return)
W:668,45: Unused argument 'is_classification' (unused-argument)

-----------------------------------
Your code has been rated at 6.37/10
```